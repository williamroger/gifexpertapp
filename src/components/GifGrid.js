import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs(category).then(imgs => setImages(imgs));
  }, [category]);


  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map(image => (
          <GifGridItem 
            key={image.id}
            {...image}
          />
        ))}
      </div>
    </>
  );
};
