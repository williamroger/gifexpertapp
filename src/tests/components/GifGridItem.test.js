import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests GifGridItem component', () => {
  const title = 'Poster Star Wars';
  const url = 'http://starwars.com/poster.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should be GifGrid', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be tag <p> has text with title', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);
  });

  test('should be tag <img> has properties url and title', () => {
    const img = wrapper.find('img');
    // console.log(img.prop('src'));
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should be tag <div> has class animate__fadeInDown', () => {
    const div = wrapper.find('div');
    const className = div.prop('className').split(' ');

    expect(className.includes('animate__fadeInDown')).toBe(true);
  });
});