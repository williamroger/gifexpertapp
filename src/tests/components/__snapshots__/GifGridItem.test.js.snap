// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests GifGridItem component should be GifGrid 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img
    alt="Poster Star Wars"
    src="http://starwars.com/poster.jpg"
  />
  <p>
    Poster Star Wars
  </p>
</div>
`;
